### PR TITLE
[ADD] Module to always show product's quatity in website

### PR DESCRIPTION
**Pull request by ayhp-odoo**
This module add functionality to always show the product quantity in the stock
in the product page of the eCommerce modifying the web controller to read the boolean field
and allows the user to still purchasing products even if the stock quantity is 0

Link to task: [2944858](https://www.odoo.com/web#id=2944858&cids=17&menu_id=4720&active_id=4532&model=project.task&view_type=form)

### CODE REVIEW

**Review by tdetry — COMMENTED** [review]
this development makes no sense in v15, right ?

**Comment by ayhp-odoo** [comment]
> Yes, just the part of displaying it on the Website

File: always_show_qty/controllers/main.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+from odoo import http
+from odoo.addons.website_sale.controllers.main import WebsiteSale
+from odoo.http import request
+
+
+class WebsiteSaleInherit(WebsiteSale):
+
+    @http.route(['/shop/<model("product.template"):product>'], type='http', auth="public", website=True, sitemap=True)
+    def product(self, product, category='', search='', **kwargs):
+        return request.render("website_sale.product", self._prepare_product_values(product, category, search, **kwargs))
```
Comment (tdetry) [inline always_show_qty/controllers/main.py:10]: not needed

File: always_show_qty/views/product_template_website_from_view_inherit.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+  <record id="product_template_form_view_inherit_website_sale_stock" model="ir.ui.view">
+    <field name="name">product.template.form.inherit.website.sale.stock.inherit</field>
+    <field name="model">product.template</field>
+    <field name="inherit_id" ref="website_sale_stock.product_template_form_view_inherit_website_sale_stock"/>
+    <field name="arch" type="xml">
+      <label for="show_availability" position="attributes">
```
Comment (tdetry) [inline always_show_qty/views/product_template_website_from_view_inherit.xml:8]: add a small comment there to mention you remove the standard behovior from the settings

File: always_show_qty/views/templates.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+  <template id="product_quantity" inherit_id="website_sale.product" active="True" name="Always Show Quatity" priority="20">
```
Comment (tdetry) [inline always_show_qty/views/templates.xml:3]: active not needed

File: always_show_qty/controllers/main.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+    def _prepare_product_values(self, product, category, search, **kwargs):
+        res = super(WebsiteSaleInherit, self)._prepare_product_values(
+            product, category, search, **kwargs)
+        if product.always_show_availability:
+            res['always_show_availability'] = True
+        return res
```
Comment (tdetry) [inline always_show_qty/controllers/main.py:17]: not needed

File: always_show_qty/controllers/main.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+
+class WebsiteSaleInherit(WebsiteSale):
+
+    @http.route(['/shop/<model("product.template"):product>'], type='http', auth="public", website=True, sitemap=True)
+    def product(self, product, category='', search='', **kwargs):
+        return request.render("website_sale.product", self._prepare_product_values(product, category, search, **kwargs))
+
+    def _prepare_product_values(self, product, category, search, **kwargs):
+        res = super(WebsiteSaleInherit, self)._prepare_product_values(
```
Comment (tdetry) [inline always_show_qty/controllers/main.py:13]: super() is enough

**Review by tdetry — CHANGES_REQUESTED** [review]

File: always_show_qty/controllers/main.py (hunk 1 above)
Comment (ayhp-odoo) [reply always_show_qty/controllers/main.py:10]: Removed all the controller for do the functionality via view template

File: always_show_qty/views/templates.xml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+  <template id="product_quantity" inherit_id="website_sale.product" name="Always Show Quatity" priority="20">
+    <xpath expr="//div[@id='o_product_terms_and_share']" position="inside">
+        <t t-if="product.always_show_availability">
+          <p class="text-success h3 mt-3">
+            <t t-esc="product.qty_available"/> <t t-esc="product.uom_id.name"/> available
+          </p>
+        </t>
+        <t t-else="" />
```
Comment (jefe-odoo) [inline always_show_qty/views/templates.xml:10]: this else node can be removed

**Pull request closed without merging**